### PR TITLE
Update VC data models to be optional

### PR DESCRIPTION
**Pull request by symorton**
**Problem:**
We need to let our data models be optional and throw an error with incorrect properties higher in the stack.

**Solution:**
We made all of the VC properties optional.

**Validation:**
All tests pass.

**Type of change:**
- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [X] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [X] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: VCEntities/VCEntities/verifiableCredential/VCClaims.swift [hunk 1]
```diff
@@ -5,13 +5,19 @@
...
+ * @see [Verifiable Credential Spec](https://www.w3.org/TR/vc-data-model/)
+ */
 public struct VCClaims: Claims {
-    public let jti: String
-    public let iss: String
-    public let sub: String
-    public let iat: Double
-    public let exp: Double
-    public let vc: VerifiableCredentialDescriptor
+    public let jti: String?
```
Comment (dangodb) [inline VCEntities/VCEntities/verifiableCredential/VCClaims.swift:14]: Should we validate the VCs before saving them in issuance? if they're missing a required property they won't be valid.

File: VCEntities/VCEntities/verifiableCredential/VCClaims.swift [hunk 2]
```diff
@@ -5,13 +5,19 @@
...
+    public let iss: String?
+    public let sub: String?
+    public let iat: Double?
+    public let exp: Double?
```
Comment (dangodb) [inline VCEntities/VCEntities/verifiableCredential/VCClaims.swift:18]: Should we add a test to make sure that we don't throw serialization exception when these properties are missing (for future regressions)

**Review by dangodb — APPROVED** [review]

File: VCEntities/VCEntities/verifiableCredential/VCClaims.swift (hunk 2 above)
Comment (symorton) [reply VCEntities/VCEntities/verifiableCredential/VCClaims.swift:18]: yes adding to next pr

File: VCEntities/VCEntities/verifiableCredential/VCClaims.swift (hunk 1 above)
Comment (symorton) [reply VCEntities/VCEntities/verifiableCredential/VCClaims.swift:14]: yes as we discussed, going to create an object validator

**Pull request merged**